tests/src/python/test_qgsgeometry.py

```diff
@@ -546,6 +547,20 @@ def testCurvePolygonPythonAdditions(self):
         with self.assertRaises(IndexError):
             g.removeInteriorRing(0)
 
+    def testOwnership(self):
+        """
+        Test ownership related exceptions
+        """
+        mp = QgsGeometry.fromWkt('MultiPoint(1 1,2 2)')
+
+        mp2 = QgsGeometry.fromWkt('MultiPoint(3 3)')
+        with self.assertRaises(OwnershipException):
+            mp2.get().addGeometry(mp.constGet().geometryN(0))
+        self.assertEqual(mp2.asWkt(), 'MultiPoint ((3 3))')
+
+        self.assertTrue(mp2.get().addGeometry(QgsPoint(3, 4)))
```

Comment:
because ""Geometry is already owned by another c++ object. Use .clone() to add a deep copy of the geometry to this multipoint." can you adapt the test to reflect this behaviour instead of adding a WKT string? This is useful o give a code guide how to do.